returnn/datasets/postprocessing.py

```diff
@@ -263,3 +289,29 @@ def _get_seq_len(self, tdict: TensorDict) -> int:
         :return: segment length of the segment in `tdict` as measured by `self.length_key` for comparison.
         """
         return tdict.data[self.length_key].raw_tensor.shape[0]
+
+
+T = TypeVar("T", TensorDict, Iterator[TensorDict])
+
+
+class Sequential(Callable):
```

Comment:
```suggestion
class Sequential(Generic[T]):
```
Do we need `Callable` here? My IDE complains about Callable not being a base class.. anyway, otherwise inherit from both.

Reply:
Right, `Callable` is not needed, this is anyway implicit.

Reply:
But I think `Generic[T]` is also not needed?

Reply:
> But I think Generic[T] is also not needed?

iirc if one does not inherit from `Generic[T]` then the T of the `__init__` could assume a different type value from the T in `__call__`

Reply:
Ah right.

Reply:
But as discussed in another thread, you would also anyway not have the `T` in the `__init__`. I.e. then also not need the `Generic[T]` here.

Reply:
Then the `T` can change from invocation to invocation, which is also not correct.

Reply:
> Then the `T` can change from invocation to invocation, which is also not correct.

There are many possible other incorrect things which are not covered by the typechecker. I don't really understand the argument.

Reply:
Note, if we want to have this type-checked, we need to use `Protocol` instead of `Callable`, but that requires Python 3.8, i.e. it depends on #1326, i.e. currently we cannot do this cleanly. I would say to just not type-check this currently, and once #1326 is solved, maybe it can be added. (But really not so important.)